server/filestore.go

```diff
@@ -5098,7 +5098,8 @@ func (fs *fileStore) PurgeEx(subject string, sequence, keep uint64) (purged uint
 					if mb.isEmpty() {
 						fs.removeMsgBlock(mb)
 						i--
-						firstSeqNeedsUpdate = seq == fs.state.FirstSeq
+						// keep flag set, if set previously
+						firstSeqNeedsUpdate = firstSeqNeedsUpdate || seq == fs.state.FirstSeq
```

Comment:
If preferred, could also update to:
```suggestion
						if !firstSeqNeedsUpdate && seq == fs.state.FirstSeq {
						    firstSeqNeedsUpdate = true
						}
```